src/L200DetectorConstruction.cc

```diff
@@ -0,0 +1,649 @@
+}
+
+void L200DetectorConstruction::InitializeMaterials(){
+	// Get nist material manager
+	G4NistManager* nist = G4NistManager::Instance();
+	G4double Z, M;
+
+   	G4double density,fractionmass,abundance;
+   	G4String name,symbol;
+   	G4int ncomponents;
+   	G4int natoms;
+   	G4double temperature, pressure;
+   	G4State  state;
+
+	//Play chemestry
+	G4Element* elH = new G4Element("H", "H", Z=1., M=1.01*g/mole);
+	//G4Element* carbon = new G4Element("C", "C", Z=6., M=12.01*g/mole);
+	//G4Element* oxygen = new G4Element("O", "O", Z=8., M=16.0*g/mole);
+	//G4Element* silicon = new G4Element("Si", "Si", Z=14., M=28.09*g/mole);
+	//G4Element* nitrogen = new G4Element("N", "N", Z=7., M=28.09*g/mole); //really this M?
+	//G4Element* elLi = new G4Element(name="Lithium",symbol="Li",Z=3.,M=6.941*g/mole);
+    	G4Element* argon = new G4Element("Argon","Ar",Z=18, M=39.948*g/mole);
+	//G4Element* elCl = new G4Element(name="Chlorine",symbol="Cl",z=17.,a=35.45*g/mole);
+	G4Element* elF = new G4Element(name="Fluorine","F",Z=9.,M=19.00*g/mole);
+	G4Element* elC = new G4Element(name="Carbon","C",Z=6.,M=12.011*g/mole);
+
+	//Some nice LAr
+  	density=1.390*g/cm3;
+  	ncomponents=1;
+  	natoms=1;
+  	state=kStateLiquid;
+  	temperature=87.0*kelvin;
+  	pressure=1.0*bar;
+  	name="LiquidArgon";
+   	lAr_mat = new G4Material(name,density,ncomponents,state,temperature,pressure);
+        lAr_mat-> AddElement(argon, natoms = 1);
+	//optical LAr
+
+	const G4int NUM = 2;
+
+
+	G4double photonEnergy[NUM] = {lambdaE/(lArWL),lambdaE/(tpbWL)};
+	G4cout << "Photonenergy";
+	G4cout << "Photon energy of " << lArWL/nm << " nm is: " <<photonEnergy[1]/eV << " eV" << G4endl;
+	G4double lArAbsorption[NUM] = {lArAbsVUV,lArAbsVis};
+	G4double lArRefIndex[NUM]={LArRefIndex(lArWL),LArRefIndex(tpbWL)};
+	G4double lArRayLength[NUM]={LArRayLength(lArWL,temperature),LArRayLength(tpbWL,temperature)};
+
+	G4MaterialPropertiesTable* mptLAr = new G4MaterialPropertiesTable();
+	mptLAr->AddProperty("RINDEX",photonEnergy,lArRefIndex,NUM);
+	mptLAr->AddProperty("RAYLEIGH",photonEnergy,lArRayLength,NUM);
+	mptLAr->AddProperty("ABSLENGTH",photonEnergy,lArAbsorption,NUM);
+
+	lAr_mat->SetMaterialPropertiesTable(mptLAr);
+
+ 	//Copper
+	copper_mat = nist->FindOrBuildMaterial("G4_Cu");
+
+	//Steel
+	steel_mat =nist->FindOrBuildMaterial("G4_STAINLESS-STEEL");
+
+	//TetraTex = PTFE
+	tetraTex_mat = new G4Material(name="tetraTex",
+				      density=2.165*g/cm3,
+				      ncomponents=2);
+  	tetraTex_mat->AddElement(elF,fractionmass= 0.76);
+  	tetraTex_mat->AddElement(elC,fractionmass= 0.24);
+
+	//TPB
+	TPB_mat = new G4Material(name="TPB",
+				 density=1.08*g/cm3,
+				 ncomponents=2);
+	TPB_mat->AddElement(elC,22);
+	TPB_mat->AddElement(elH,28);
+	//optical TPB
+	G4double tpbQuantumEff = 1.2;
+	G4double tpbTimeConst = 0.01*ns;
+	G4double tpbRefIndex[NUM] = {1.635,1.635};
+	G4double tpbEmission[NUM] = {lambdaE/tpbWL,lambdaE/tpbWL};
+	G4double tpbAbsorption[NUM] = {1*nm, 1000*m};
```

Comment:
Think there are some errors here (giving me energy-nonconservation & ~weird UV-Emissions, see example.txt).
1) G4 interpolates emission/absorption; I think we would like to have 2 delta-distr. peaks for abs. & emiss., not a linear transition
2) emission seems wrong. I think we would like to have 1. for emiss.peak and 0. for everything else.
I would take over this, if that is fine for you